controllers/storageclaim_controller.go

```diff
@@ -321,6 +321,24 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 					return reconcile.Result{}, fmt.Errorf("failed to extract monitor data: %v", err)
 				}
 				csiClusterConfigEntry.Monitors = append(csiClusterConfigEntry.Monitors, monitorIps...)
+			} else if eResource.Kind == "ConfigMap" && eResource.Name == csiSidecarConfigMapName {
+				csiSideCarConfig := &corev1.ConfigMap{}
+				csiSideCarConfig.Name = eResource.Name
+				csiSideCarConfig.Namespace = r.OperatorNamespace
+
+				data := map[string]string{}
+				err = json.Unmarshal(eResource.Data, &data)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshal csi-sidecar-config configuration response: %v", err)
+				}
+
+				_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, csiSideCarConfig, func() error {
```

Comment:
how do you intend to clean this up? Is it missing owner ref? Should CM be the owner as CM (~CSI) lifetime is more than StorageClaim. What happens if one clients says `yes` and another `no`? Is that possible?